### PR TITLE
Expose function_calls in chat block response

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR modifies our chat block's response to include `function_calls`. This change is required as models can now use multiple functions in a single call. For backward compatibility, we keep the legacy `function_call` field, which corresponds to the first function used. It adds the `id` field to both implementations.

A follow-up PR is required to accept `function_calls` in the history of messages.

**Example:**

image

### CODE REVIEW

File: core/src/providers/anthropic.rs [hunk 1]
```diff
@@ -115,31 +127,6 @@ impl TryFrom<AnthropicStreamContent> for AnthropicResponseContent {
     }
 }
 
-impl TryFrom<&AnthropicResponseContent> for ChatMessage {
```
Comment (flvndvd) [inline core/src/providers/anthropic.rs:118]: This was not used.

**Review by spolu — APPROVED** [review]
Looks great. 1 change request. Feel free to merge once done :+1:

File: core/src/providers/mistral.rs [hunk 2]
```diff
@@ -121,6 +121,18 @@ impl TryFrom<&ChatFunctionCall> for MistralToolCall {
     }
 }
 
+impl TryFrom<&MistralToolCall> for ChatFunctionCalls {
+    type Error = anyhow::Error;
+
+    fn try_from(tc: &MistralToolCall) -> Result<Self, Self::Error> {
+        Ok(ChatFunctionCalls {
+            id: new_id(),
```
Comment (spolu) [inline core/src/providers/mistral.rs:129]: Let's add a prefix here `fc_`

**Review by spolu — APPROVED** [review]
Another small change request

File: core/src/providers/mistral.rs (hunk 2 above)
Comment (flvndvd) [reply core/src/providers/mistral.rs:129]: Done.

File: core/src/providers/llm.rs [hunk 3]
```diff
@@ -76,6 +83,8 @@ pub struct ChatMessage {
     pub content: Option<String>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub function_call: Option<ChatFunctionCall>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub function_calls: Option<Vec<ChatFunctionCalls>>,
```
[the remark this answers is not in the document]
Comment (flvndvd) [reply core/src/providers/llm.rs:87]: Ok, let me refactor then.

[the remark this answers is not in the document]
Comment (flvndvd) [reply core/src/providers/llm.rs:87]: Done.

**Comment by spolu** [comment]
:100:

**Pull request merged**